### PR TITLE
graphql-upload: Update types to v16

### PR DESCRIPTION
**Pull request by fmauNeko**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jaydenseric/graphql-upload/releases/tag/v16.0.0
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by rdsfj** [comment]
Hey @fmauNeko, thanks for your MR!
There are some merge conflicts right now, possibly because of this other MR over here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64515.
Can you fix them?

Apart from that, the changes look good to me 👍 Unfortunately, I can't approve because there are changes in the package.json. One of the maintainers needs to approve.

**Comment by fmauNeko** [comment]
> Hey @fmauNeko, thanks for your MR! There are some merge conflicts right now, possibly because of this other MR over here: #64515. Can you fix them?
> 
> Apart from that, the changes look good to me 👍 Unfortunately, I can't approve because there are changes in the package.json. One of the maintainers needs to approve.

Hey !
That is fixed, there were other changes in #64516, I rebased on latest `master`.

File: types/graphql-upload/package.json [hunk 1]
```diff
@@ -6,20 +6,20 @@
   },
   "exports": {
     "./package.json": "./package.json",
-    "./GraphQLUpload.js": {
-      "types": "./GraphQLUpload.d.ts"
+    "./GraphQLUpload.mjs": {
```
Comment (jakebailey) [inline types/graphql-upload/package.json:9]: I'm shocked by this, but, the upstream package really _does_ put extensions in its export map. Very odd. Can't be intended...

Comment (jakebailey) [reply types/graphql-upload/package.json:9]: I guess it is. Weird. https://github.com/jaydenseric/graphql-upload/releases/tag/v16.0.0

**Review by jakebailey — APPROVED** [review]

**Pull request merged**